custom-elements/CustomElementRegistry.html

```diff
@@ -721,17 +723,24 @@
         assert_false('rejected' in promiseAfterDefine, 'The promise returned by "whenDefined" must not be rejected until the end of the next microtask');
     }).then(function () {
         assert_true('resolved' in promise, 'The promise returned by "whenDefined" must be resolved when a custom element is defined');
-        assert_equals(promise.resolved, undefined,
-            'The promise returned by "whenDefined" must be resolved with "undefined" when a custom element is defined');
+        assert_equals(promise.resolved, ElementDefinedAfterWhenDefined,
+            'The promise returned by "whenDefined" must be resolved with the constructor of the element when a custom element is defined');
         assert_false('rejected' in promise, 'The promise returned by "whenDefined" must not be rejected when a custom element is defined');
 
         assert_true('resolved' in promiseAfterDefine, 'The promise returned by "whenDefined" must be resolved when a custom element is defined');
-        assert_equals(promiseAfterDefine.resolved, undefined,
-            'The promise returned by "whenDefined" must be resolved with "undefined" when a custom element is defined');
+        assert_equals(promiseAfterDefine.resolved, ElementDefinedAfterWhenDefined,
+            'The promise returned by "whenDefined" must be resolved with the constructor of the element when a custom element is defined');
         assert_false('rejected' in promiseAfterDefine, 'The promise returned by "whenDefined" must not be rejected when a custom element is defined');
     });
 }, 'A promise returned by customElements.whenDefined must be resolved by "define"');
 
+promise_test(function () {
+    class ResolvedCustomElement extends HTMLElement {};
+    customElements.define('resolved-custom-element', ResolvedCustomElement);
+    return customElements.whenDefined('resolved-custom-element').then(function (value) {
```

Comment:
We should another test case where we call whenDefined before define is called.

Reply:
Done